src/main/java/com/mineboundteam/minebound/magic/helper/UseSpellHelper.java

```diff
@@ -12,32 +13,50 @@ public class UseSpellHelper {
     public static void useSpell(Player player, Capability<? extends SelectedSpell> cap) {
         player.getCapability(cap).ifPresent(selected -> {
             if (!selected.isEmpty()) {
-                ItemStack activeSpell = ItemStack.of(ArmorNBTHelper.getSpellTag(player.getItemBySlot(selected.equippedSlot), ArmorNBTHelper.ACTIVE_SPELL).getCompound(selected.index));
+                ItemStack armor = player.getItemBySlot(selected.equippedSlot);
+                ListTag tagList = ArmorNBTHelper.getSpellTag(armor, ArmorNBTHelper.ACTIVE_SPELL);
+
+                ItemStack activeSpell = ItemStack.of(tagList.getCompound(selected.index));
                 if (activeSpell.getItem() instanceof ActiveSpellItem spell) {
                     spell.use(activeSpell, player.level, player);
                 }
+
+                tagList.set(selected.index, activeSpell.serializeNBT());
+                ArmorNBTHelper.saveSpellTag(armor, ArmorNBTHelper.ACTIVE_SPELL, tagList);
```

Comment:
Are these changes to fix the issue I just made today?

Reply:
No, I don't think so. There was an issue where, when I was editing the nbt data of the spell item while it was equipped on the armor, it wasn't saving.

Reply:
Oh, I see.